portfolio/src/main/java/com/google/sps/servlets/SuggestServlet.java

```diff
@@ -100,4 +116,41 @@ private String getParameter(HttpServletRequest request, String name, String defa
     }
     return value;
   }
+
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    // BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    // if (blobInfo.getSize() == 0) {
+    //   blobstoreService.delete(blobKey);
+    //   return null;
+    // }
+
+    // We could check the validity of the file here, e.g. to make sure it's an image file
+    // https://stackoverflow.com/q/10779564/873165
```

Comment:
Looks like you might be able to use ImagesService to check the format of the uploaded blob and delete it if it's not an image
https://stackoverflow.com/questions/12549752/how-to-prevent-non-images-from-being-uploaded